Patch-Source: https://github.com/chimera-linux/cports/blob/ee0ee90ac5d7ddd2466e4fdd40d1ebd0c22467f4/main/libsoup/patches/unsigned-char.patch
--
commit d4b2862b663a85903b688c7e2eae5aa0b551a14e
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Sat Mar 18 01:06:12 2023 +0100

    use int for env_force_http1
    
    As char is unsigned on some targets, this would result in
    env_force_http1 always being truthy.

diff --git a/libsoup/soup-connection-manager.c b/libsoup/soup-connection-manager.c
index 981a3d3..e6f7caa 100644
--- a/libsoup/soup-connection-manager.c
+++ b/libsoup/soup-connection-manager.c
@@ -378,7 +378,7 @@ static SoupConnection *
 soup_connection_manager_get_connection_locked (SoupConnectionManager *manager,
                                                SoupMessageQueueItem  *item)
 {
-        static gchar env_force_http1 = -1;
+        static int env_force_http1 = -1;
         SoupMessage *msg = item->msg;
         gboolean need_new_connection;
         SoupConnection *conn;
